source/features/index.tsx

```diff
@@ -82,20 +83,41 @@ let logError = (id: FeatureID, error: unknown): void => {
 
 // eslint-disable-next-line no-async-promise-executor -- Rule assumes we don't want to leave it pending
 const globalReady: Promise<RGHOptions> = new Promise(async resolve => {
-	await elementReady('body', {waitForChildren: false});
+	const [options, hotfix, bisectedFeatures] = await Promise.all([
+		optionsStorage.getAll(),
+		version === '0.0.0' || await cache.get('hotfix'), // Ignores the cache when loaded locally
+		bisectFeatures()
+	]);
 
-	if (pageDetect.is500()) {
-		return;
+	if (options.customCSS.trim().length > 0) {
+		document.head.append(<style>{options.customCSS}</style>);
 	}
 
-	if (document.title === 'Confirm password' || document.title === 'Confirm access') {
-		return;
+	if (bisectedFeatures) {
+		Object.assign(options, bisectedFeatures);
+	} else {
+		// If features are remotely marked as "seriously breaking" by the maintainers, disable them without having to wait for proper updates to propagate #3529
+		void checkForHotfixes();
+		Object.assign(options, hotfix);
 	}
 
-	if (document.body.classList.contains('logged-out')) {
-		console.warn('Refined GitHub is only expected to work when you’re logged in to GitHub. Errors will not be shown.');
-		features.error = () => {/* No logging */};
-		logError = () => {/* No logging */};
+	// Create logging function
+	log = options.logging ? console.log : () => {/* No logging */};
+
+	await Promise.race([
+		// With comments
+		elementReady('body', {waitForChildren: false}),
```

Comment:
@fregante why do we need this at all?

Reply:
rAF might still be called earlier than setTimeout, in my mind. 

I wonder if we should just use oneMutation on `document.documentElement` to wait for the body. This would avoid any timers

Reply:
No freeze on firefox but takes 30 - 50ms

Reply:
No freeze on firefox but takes 30 - 50ms